includes/class-duplicate-payment-prevention-service.php

```diff
@@ -115,7 +115,28 @@ public function check_payment_intent_attached_to_order_succeeded( WC_Order $orde
 		if ( Intent_Status::SUCCEEDED === $intent_status ) {
 			$this->remove_session_processing_order( $order->get_id() );
 		}
-		$this->order_service->update_order_status_from_intent( $order, $intent );
+
+		// Check if the order amount matches the charged amount.
+		$order_total_in_cents = (int) ( $order->get_total() * 100 );
```

Comment:
Multiplying by 100 is not the right approach, see https://github.com/Automattic/woocommerce-payments/blob/1621f846a667bce6b233b1faedb2fdf9d5bd73bd/includes/class-wc-payment-gateway-wcpay.php#L1511
```suggestion
		$order_total_in_cents = WC_Payments_Utils::prepare_amount( $amount, $order->get_currency() )
```

Reply:
thank you for calling that out. addressed: cbbbd7d1fcc51e47b6b4851540238b3b8a797a63